src/classify.jl

```diff
@@ -16,3 +16,44 @@ function classify!(array, weights, mask = nothing)
     end
     array
 end
+
+"""
+    blend(arrays[, scaling])
+
+Blend arrays weighted by scaling factors.
+"""
+function blend(arrays, scaling = ones(length(arrays)))
+    ret = sum(arrays .* scaling)
+    rescale!(ret)
+end
+
+function _clusterMean(clusterArray, array)
+    clusters = Dict{Float64, Float64}()
+    clustersum = Dict{Float64, Float64}()
+    for ind in eachindex(clusterArray, array)
+        temp = clusterArray[ind]
+        if !haskey(clusters, temp)
+            clusters[temp] = clustersum[temp] = 0.0
+        end
+        clusters[temp] += 1.0
+        clustersum[temp] += array[ind]          
+    end
+    for cl in keys(clusters)
+        clustersum[cl] /= clusters[cl]
+    end
+    clustersum[NaN] = NaN
+    _rescale!(get.(Ref(clustersum), clusterArray, NaN))
```

Comment:
what is the `_rescale!` `rescale!` difference?

Reply:
there is no rescale! function

Reply:
I updated this now